src/ecs/systems/rendering.rs

```diff
@@ -54,19 +61,47 @@ impl<'a> System<'a> for RenderSystem {
     }
 }
 
-impl RenderSystem {
+impl<'a, 'b> SystemExt<'a, (&'b EventsLoop, PipelineBuilder)> for RenderSystem {
```

Comment:
`SystemExt` to be used with `Application`.
`Application::with_system` will require `S: for<'a> SystemExt<'a>`, it will accept building arguments for it and build it in place providing `World` to register any relevant `Component`s and put `Resource`s